src/org/pentaho/reporting/platform/plugin/JobManager.java

```diff
@@ -237,6 +247,24 @@ public Response schedule( @PathParam( "job_id" ) final String jobId ) {
     return Response.ok().build();
   }
 
+  @GET @Path( "{job_id}/{folderId}/updateLocation" ) @Produces( "text/text" )
+  public Response updateLocation( @PathParam( "job_id" ) final String jobId,
+                                  @PathParam( "folderId" ) final String folderId ) {
+    final IPentahoAsyncExecutor executor = getExecutor();
+    final IPentahoSession session = PentahoSessionHolder.getSession();
+    final UUID uuid;
+    try {
+      uuid = UUID.fromString( jobId );
+    } catch ( final Exception e ) {
+      logger.error( INVALID_UUID + jobId );
+      return get404();
+    }
+
+    executor.updateSchedulingLocation( uuid, session, folderId );
```

Comment:
You never check whether the jobId is actually valid. I can spam you with infinite syntactically correct UUIDs and you will happily fill up the location map. (The executor implementation does not check the UUID either. 

Second: REST principles state that if you want to modify state on the server, you never ever use a GET method for that. GETs may be cached by a proxy, and are generally assumed to be read only operations. 

I would also say that the folder ID should probably not be a path parameter. In the context of this operation is does not make much sense to present the folders as tree/path structure. Use a normal form parameter instead.

/{job_id}/updateLocation?folderId=/my/path/here.

If you really want to use a path parameter, I would swap folderId and the "updateLocation" key, as "updateLocation" is an operation on the job, not on the path.